### DISCUSSION TITLE
Markdown: Syntax Highlighter for C# has become very buggy

### DISCUSSION THREAD

**Discussion by thygrrr** [community/community#128273] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

The csharp syntax highlighter, e.g. for use in README.md, doesn't even recognize the "var" keyword anymore; it does not color named arguments, etc.

This is true for `csharp`and `cs`. Out of these, the most egregious is clearly the `var` keyword being uncolored, but also many other language features are interpreted poorly. With C# 13 around the corner that adds a host of additional keywords, a refresh, review, and repair of the highlighter is urgently needed.

The faulty highlighting appears unrelated to the github theme used.

## Sample code:

```cs
// Declare a component record. (we can also use most existing value & reference types)
record struct Velocity(Vector3 Value);

// Create a world. (fyi, World implements IDisposable)
var world = new fennecs.World();

// Spawn an entity into the world with a choice of components. (or add/remove them later)
var entity = world.Spawn().Add<Velocity>();

// Queries are cached & we use ultra-lightweight Stream Views to feed data to our code!
var stream = world.Query<Velocity>().Stream();

// Run code on all entities in the query. (exchange 'For' with 'Job' for parallel processing)
stream.For(
    uniform: DeltaTime * 9.81f * Vector3.UnitZ,
    action: (Vector3 uniform, ref Velocity velocity) =>
    {
        velocity.Value -= uniform;
    }
);
```

## Sample output:
![image](https://github.com/community/community/assets/8904620/674d5495-5b06-46a5-9b3a-dda124165cf2)

## Expected output:
![image](https://github.com/community/community/assets/8904620/2cdca9d6-b0f1-454d-8b5d-d4433eed6dab)

**Comment by spenserblack** [comment]
I believe that GitHub's Markdown renderer uses [the grammars provided by Linguist](https://github.com/github-linguist/linguist/tree/master/vendor/grammars). Check out [their note on their grammers](https://github.com/github-linguist/linguist/blob/42fd3c2514375fc6ad281552368edd3fb9f6ee7f/vendor/README.md) -- C# is marked with a :snail: which means it isn't updated as frequently as other grammars.

**Comment by thygrrr** [comment]
Thanks, I was unable to determine what highlighter they even use. I'll leave an issue there. .)
